### ISSUE TITLE
TCP, UDP, WS / Networking brief overview

### ISSUE THREAD

**Issue by gzhu725** [gzhu725/gzhu725#23]
# [TCP/IP OV](https://www.youtube.com/watch?v=PpsEaqJV_A0)
- talks about TCP only
- TCP is a cake with layers: network,internet, transmission, application (from bottom to top)
- application: has direct protocols (like smtp, HTTP)
- transport: TCP or UDP, talks to this layer via ports (diff protocol has a diff port)
- TCP chunks everything in packets and then goes to wherever it needs to go 
- TCP has a header with instructions for what order to reassmble packets and error tracking for no packet errors
- then, packets are pushed onto INTERNET LAYER , uses IP to attach origin and destination IP addresses so packet kows where it came from and where to go
- network: mac addresses, streamlines process (don't have to bottleneck sending...little traffic)

# [TCP vs UDP](https://www.youtube.com/watch?v=cA9ZJdqzOoU)
- transport layer: TCP and UDP, how do you send your data? reliability or unreliable (+ and -)
- TCP: transmission control protocol 
- reliable transmitting of data. will check for errors and recover! 
- uses acknowledgements, sequencing, and check sum to check for errors (checksum == checksum on both sides, sequential order matters and to acknowledge)
- UDP: real time fast no latency and its fast. like voice calls, video gaming, etc. where we want REAL TIME STUFF 
- less info than TCP....doesn't care for drops. 

# WS 
- works like TCP , works on top of TCP 
- basically some other protocol lol
